### PR TITLE
refactor: run relay runtime in async task group

### LINKED ISSUE #11: Add horizontal mouse wheel aka pan
As of now, only "regular" vertical scrolling is supported.

Horizontal scrolling would be desirable.

### LINKED ISSUE #172: Simplify relay device add/remove lifecycle
Follow-up from PR #171 review.

The relay controller's device add/remove path works, but the lifecycle code has become harder to reason about after hotplug, cancellation, and cleanup handling were split across queued startup adds, active tasks, active device handles, and task finalizers.

Goals:

- Revisit whether the active-device and active-task bookkeeping can be simplified.
- Keep cancellation and `InputDevice.close()` ownership explicit.
- Preserve current behavior for startup queues, hotplug remove events, and disappeared devices.
- Add focused tests for the simplified lifecycle before changing runtime behavior.

### PR DESCRIPTION
**Pull request by quaxalber**
## Summary

- Splits relay runtime code into focused modules for gadget management, relay control, device relay, HID dispatch, runtime monitoring, shortcut toggling, and device identification.
- Keeps `bluetooth_2_usb.evdev` as the stable facade while moving bundled Linux input constants into `evdev_ecodes.py`.
- Moves mouse delta coalescing and HID report chunking into `mouse_delta.py` with shared HID bounds.
- Adds extended mouse report support for 16-bit X/Y motion, pan, wheel, and one byte of button state.
- Standardizes package logging on named module loggers while keeping handlers on the package logger.
- Hardens shutdown, hotplug, device handle ownership, relay failure propagation, gadget release, transient UDC reads, loopback settle validation, identifier validation, and relay restart edge cases.

## Issues

Closes #11.
Closes #172.
Refs #170.

## Validation

- `venv/bin/black --check src tests`
- `venv/bin/ruff check src tests`
- `venv/bin/python -m compileall src tests`
- `venv/bin/python -m unittest discover -s tests -v` (155 tests)
- `bash -n` for public shell scripts under `scripts/`
- Public shell script `--help` checks under `scripts/`
- Python CLI `--help` and `--version`
- Doc consistency review per `docs/doc-consistency-review.md`; no doc fixes were needed because the current changes do not alter public CLI flags, script names, managed paths, or operator commands.
- Pi 4B live validation on current head: deployed to `/opt/bluetooth_2_usb`, restarted `bluetooth_2_usb.service`, verified service active, and passed loopback `combo` pinned to fresh node `1-2.1.2`.
- Pi 4B live validation on the preceding lifecycle-refactor head also passed loopback `keyboard`, `mouse`, `mouse_fast`, `combo`, and `consumer` pinned to fresh node `1-2.1.2`.
- Earlier Pi 4B validation on this PR included `smoketest.sh --verbose`, `debug.sh --duration 10`, `bluetoothctl show`, `btmgmt info`, and loopback `keyboard`, `mouse`, `mouse_fast`, `combo`, and `consumer`.

## Summary by CodeRabbit

* **New Features**
  * Enhanced mouse support including wheel, pan, and extended button handling with hi-res detection
  * Graceful shutdown mechanism with configurable timeout for relay operations
  * Loopback injector now respects service settle window timing via `B2U_LOOPBACK_SERVICE_SETTLE_SEC` environment variable
  
* **Improvements**
  * Improved device hotplug handling with retry logic for transient failures
  * Better error recovery for HID operations with incremental backoff
  * Enhanced documentation clarifying Windows input backend limitations
  * Modular architecture supporting better testability and maintainability

### CODE REVIEW

**Review by quaxalber — COMMENTED** [review]
[old review comment removed]

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

File: src/bluetooth_2_usb/device_relay.py [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+                if isinstance(event, RelEvent):
+                    self._accumulate_mouse_movement(event)
+                    continue
+
+                if is_syn_report:
+                    await self._flush_pending_mouse_movement()
+                    continue
+
+                await self._flush_pending_mouse_movement()
```
Comment (quaxalber) [inline src/bluetooth_2_usb/device_relay.py:180]: Is it correct to flush here without syn?

File: src/bluetooth_2_usb/gadget_manager.py [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+        return self._gadgets["mouse"]
+
+    def get_consumer(self) -> ConsumerControl | None:
+        """
+        Get the ConsumerControl gadget.
+
+        :return: A ConsumerControl object, or None if not initialized
+        :rtype: ConsumerControl | None
+        """
+        return self._gadgets["consumer"]
```
Comment (quaxalber) [inline src/bluetooth_2_usb/gadget_manager.py]: Convert to properties

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review

**Comment by quaxalber** [comment]
@coderabbitai full review
